packages/apputils/src/widgettracker.ts

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { InstanceTracker } from '@jupyterlab/coreutils';
+
+import { IDisposable } from '@phosphor/disposable';
+
+import { ISignal, Signal } from '@phosphor/signaling';
+
+import { FocusTracker, Widget } from '@phosphor/widgets';
+
+/**
+ * A tracker that tracks widgets.
+ *
+ * @typeparam T - The type of widget being tracked. Defaults to `Widget`.
+ */
+export interface IWidgetTracker<T extends Widget = Widget> extends IDisposable {
```

Comment:
In `coreutils` you have put the interface for `IInstanceTracker` in `instances.ts`. Any reason to not do the same here?

Reply:
`apputils` does not have an `interfaces.ts` file. Do you think it should have one? Since `coreutils` already had this pattern, I wanted to adhere to it.

Reply:
I don't really have strong opinions on this one.